### PR TITLE
Create doc-proposal.md to capture the new doc proposal for AWX-Operator

### LINKED ISSUE #1360: Split Readme up into multiple files
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx-operator/issues) for duplicates.
- [X] I understand that AWX Operator is open source software provided for free and that I might not receive a timely response.

### Feature Summary

In preparation for creating a docsite, split up the existing readme into multiple smaller files in the /docs folder.

- [x] draft out a proposal for how to split the file up. Use a comment in this github issue to describe the proposed set of new files.
- [x] Once the writers agree, move that information into the agreed list of files.
- [x] Leave everything from Contributing section to the end of the README file in the README file.
- [x] Leave everything BEFORE the Table of contents in the README file.
- [ ] Add the new files to the mkdocs.yml and test the docs build

### PR DESCRIPTION
**Pull request by DesmondSanctity**
##### SUMMARY
This commit contains the addition of the markdown file for the new awx-operator doc proposal

##### ISSUE TYPE
 - Bug, Docs Fix or other nominal change

##### ADDITIONAL INFORMATION
This PR fixes #1360

### CODE REVIEW

File: docs/doc-proposal.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
...com/ansible/awx-operator#session-cookie-secure-setting)<br>    * [Extra Settings](https://github.com/ansible/awx-operator#extra-settings)<br>    * [Configure no_log](https://github.com/ansible/awx-operator#no-log)<br>    * [Auto Upgrade](https://github.com/ansible/awx-operator#auto-upgrade)<br>        ** [Upgrade of instances without auto upgrade](https://github.com/ansible/awx-operator#upgrade-of-instances-without-auto-upgrade)<br>    * [Service Account](https://github.com/ansible/awx-operator#service-account)<br>    * [Labeling operator managed objects](https://github.com/ansible/awx-operator#labeling-operator-managed-objects)<br>    * [Pods termination grace period](https://github.com/ansible/awx-operator#pods-termination-grace-period)<br>    * [Disable IPV6](https://github.com/ansible/awx-operator#disable-ipv6)<br>    * [Add Execution Nodes](https://github.com/ansible/awx-operator#adding-execution-nodes)<br>        ** [Custom Receptor CA](https://github.com/ansible/awx-operator#custom-receptor-ca)<br>    * [Debugging](https://github.com/ansible/awx-operator/blob/devel/docs/debugging.md)<br>    * [Migration](https://github.com/ansible/awx-operator/blob/devel/docs/migration.md) |
```
Comment (rooftopcellist) [inline docs/doc-proposal.md:23]: FYI @DesmondSanctity , it looks like the nested bullet points don't show up indented as you would expect when rendered in GH markdown.  For example:
image

Comment (rooftopcellist) [reply docs/doc-proposal.md:23]: Oh, I see your comment below:
> Note: I could not get the multi-level bullet point list to work in the table so I used single asterisk `*` for one level down and double asterisk `**` for two level down.

Comment (DesmondSanctity) [reply docs/doc-proposal.md:23]: Yes. It wouldn't work inside the table in GitHub markdown for some reasons I don't know.

So I had to add the note. Thanks for the review

Comment (rooftopcellist) [reply docs/doc-proposal.md:23]: Passing idea, you could do something like this to achieve the  indent:
```
- [Admin user account configuration](https://github.com/ansible/awx-operator#admin-user-account-configuration)
- [Network and TLS Configuration](https://github.com/ansible/awx-operator#network-and-tls-configuration)
--- [Service Type](https://github.com/ansible/awx-operator#service-type)
--- [Ingress Type](https://github.com/ansible/awx-operator#ingress-type)
```

Neither option is ideal, but it's not a big deal either way.

Comment (DesmondSanctity) [reply docs/doc-proposal.md:23]: Okay. I will try it and see.

**Review by rooftopcellist — APPROVED** [review]

**Comment by rooftopcellist** [comment]
Thank you for putting this together @DesmondSanctity !

It will be merged shortly.

**Pull request merged**